### PR TITLE
FF-2945 Add announcement for Python SDK 4.0

### PR DESCRIPTION
**Pull request by rasendubi**
To be merged after https://github.com/Eppo-exp/rust-sdk/pull/29

### CODE REVIEW

**Comment by rasendubi** [comment]
cc @aarsilv @leoromanovsky

File: docs/sdks/server-sdks/python/index.md
Comment (rasendubi) [inline docs/sdks/server-sdks/python/index.md:1]: The changes in this file are just automatic trailing lines removes

File: docs/sdks/server-sdks/python/announcing-python-4-0.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+Using a shared core library allows us to centralize our efforts and provide better performance, stability, and deliver new features faster.
+
+## Better performance
+
+Thanks to Rust’s natural speed, the assignment evaluation performance improved 2–5x depending on the feature flag configuration.
+You may expect even better performance in the future as we continue to optimize the core library.
+
+Additionally, configuration update now happens in a background thread without ever holding Global Interpreter Lock (GIL), so it runs in parallel with your Python threads without slowing them down.
+
```
Comment (aarsilv) [inline docs/sdks/server-sdks/python/announcing-python-4-0.md:16]: I wonder if this is a good spot to put the default polling interval has been changed from 300 seconds to 30

**Review by aarsilv — APPROVED** [review]
💪  EXCITING 🚀

File: docs/sdks/server-sdks/python/announcing-python-4-0.md (hunk 1 above)
Comment (leoromanovsky) [reply docs/sdks/server-sdks/python/announcing-python-4-0.md:16]: Thanks for the reminder! Aaron is referring to this: 

https://github.com/Eppo-exp/python-sdk/blob/main/eppo_client/constants.py

**Pull request merged**